### PR TITLE
Make padding optional

### LINKED ISSUE #483: Make timestamp padding optional
**Is your feature request related to a problem? Please describe.**
The component `@bbc/psammead-timestamp` always has padding. This causes issues for us on the ws frontpages as we use the component in the story promo

image

**Describe the solution you'd like**
Make timestamp padding optional via a prop, this will likely take the form of
```
<Timestamp padding={false} />
```
If `props.padding` is false, remove all padding. If it's true, add the padding currently implemented. 

The prop `padding` should default to `true`

**Describe alternatives you've considered**
Implementing it like
```
import Timestamp, { TimestampNoPadding } from @bbc/psammead-timestamp
```

**Testing notes**
[Tester to complete]

Dev insight: Snapshots and a storybook knob boolean

- [x] Initially labelled with ["Refinement needed"](https://github.com/bbc/psammead/labels/Refinement%20Needed)

### PR DESCRIPTION
**Pull request by ghost**
Resolves #483 

**Overall change:** Introduced a `padding` prop for padding control.
**Code changes:**
- Added `padding` prop and made it optional and defaults to `true`.
- Added padding knob to storybook
- Wrote a test case for `Timestamp` without padding.

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Tests added for new features
- [ ] Test engineer approval

### CODE REVIEW

File: packages/components/psammead-timestamp/src/index.jsx [hunk 1]
```diff
@@ -15,17 +15,22 @@ const StyledTimestamp = styled.time`
...
   display: block;
   font-family: ${GEL_FF_REITH_SANS};
-  padding-bottom: ${GEL_SPACING_HLF};
-  &:last-child {
-    padding-bottom: ${GEL_SPACING_DBL};
-  }
-`;
+  ${props =>
+    props.padding &&
+    css`
```
Comment (dr3) [inline packages/components/psammead-timestamp/src/index.jsx]: This a bit tricky to parse, can you pull this out into a variable and then use it here please

**Review by dr3 — CHANGES_REQUESTED** [review]

**Review by dr3 — DISMISSED** [review]
LGTM :)

**Review by j-pendlebury — DISMISSED** [review]
👍

**Review by dr3 — APPROVED** [review]

File: packages/components/psammead-timestamp/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@bbc/psammead-timestamp",
-  "version": "0.2.0",
+  "version": "0.3.1",
```
Comment (sareh) [inline packages/components/psammead-timestamp/package.json:3]: I just noted that this other PR did not increment the version number https://github.com/bbc/psammead/pull/494#issuecomment-490192237 and so version 0.3.0 is merged on `latest` but not published. 

So can you please hold off on merging this PR until that fix has been addressed? Thanks.

**Review by andrew-nowak — COMMENTED** [review]
There are merge conflicts, otherwise lgtm

**Review by andrew-nowak — APPROVED** [review]

**Comment by jamesbrumpton** [comment]
LGTM. Happy for this to be merged.

**Pull request merged**